### PR TITLE
Implement nullish coalescing operator in parser

### PR DESCRIPTION
**Pull request by azz**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | yes
| Deprecations?     | no
| Spec compliancy?  | yes
| Tests added/pass? | yes
| Fixed tickets     | 
| License           | MIT

Adding the parser for the nullish coalescing operator, `a ?? b`, currently at stage 1.

In the AST it isn an operator under `BinaryExpression`. Should it move to `LogicalExpression`?

Picked the plugin name: `"nullishCoalescingOperator"`. It's a bit of a mouthful, and may well change.

Ref: https://github.com/babel/proposals/issues/14 
Proposal: https://github.com/gisenberg/proposal-null-coalescing

Proposal will likely be renamed from `nullary` to `nullish`, see https://github.com/gisenberg/proposal-null-coalescing/issues/3.

I feel like I haven't added enough tests, but I'm not too sure of the edge cases I should test, let me know if you know of any!

### CODE REVIEW

**Comment by hzoo** [comment]
For the tests, I would check https://github.com/babel/babylon/pull/742/files which is similar

Could do chained so `a ?? b ?? c` or multiline, or precedence check

**Comment by gisenberg** [comment]
cc @littledan

**Comment by hzoo** [comment]
Fine with merging now if we can get a PR for tests next, (logic shouldn't be any different)

**Comment by hzoo** [comment]
new PR https://github.com/babel/babylon/pull/762, should probably be `LogicalExpression` instead of `BinaryExpression`

File: ast/spec.md [hunk 1]
```diff
@@ -791,6 +791,7 @@ enum BinaryOperator {
      | "|" | "^" | "&" | "in"
      | "instanceof"
      | "|>"
+     | "??"
```
Comment (jridgewell) [inline ast/spec.md]: This is a `LogicalOperator`, not a `BinaryOperator`

Comment (hzoo) [reply ast/spec.md]: https://github.com/babel/babylon/pull/762
